core/tx_list.go

```diff
@@ -210,12 +223,27 @@ func (m *txSortedMap) Flatten() types.Transactions {
 		}
 		sort.Sort(types.TxByNonce(m.cache))
 	}
+	return m.cache
+}
+
+// Flatten creates a nonce-sorted slice of transactions based on the loosely
+// sorted internal representation. The result of the sorting is cached in case
+// it's requested again before any modifications are made to the contents.
+func (m *txSortedMap) Flatten() types.Transactions {
 	// Copy the cache to prevent accidental modifications
-	txs := make(types.Transactions, len(m.cache))
-	copy(txs, m.cache)
+	cache := m.flatten()
+	txs := make(types.Transactions, len(cache))
+	copy(txs, cache)
 	return txs
 }
 
+// LastElement returns the last element of a flattened list, thus, the
+// transaction with the highest nonce
+func (m *txSortedMap) LastElement() *types.Transaction {
+	cache := m.flatten()
+	return cache[len(cache)-1]
```

Comment:
I know that this is probably from upstream, but allocate space, and then sort the list, to just get the biggest nonce, seems like a lot of unnecessary work... Maybe we could create an issue for this. Maybe we are expecting to have only a few elements per list and I'm overreacting, don't know

Reply:
I'm still in the process of reviewing this change from upstream, but it seems like they expect more reads than writes w/ their caching strategy.

Reply:
I usually trust upstream judgment on these things, but we could definitely take a look.